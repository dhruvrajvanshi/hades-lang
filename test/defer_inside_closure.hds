
def main(): Void {
  val m = b"DEFER_OUTSIDE_CLOSURE"
  defer puts(m)
  puts(b"BEFORE_APPLY")
  apply || {
    val m = b"DEFER_INSIDE_CLOSURE"
    defer puts(m)
    puts(b"BEFORE_DEFER_INSIDE_CLOSURE")
    return
  }

  puts(b"AFTER_APPLY")
}
extern def puts(*u8): Void = puts

def apply(f: || -> Void): Void {
    f()
}