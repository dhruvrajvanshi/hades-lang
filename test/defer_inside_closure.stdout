BEFORE_APPLY
BEFORE_DEFER_INSIDE_CLOSURE
DEFER_INSIDE_CLOSURE
AFTER_APPLY
DEFER_OUTSIDE_CLOSURE
